src/pruneOrGetComponentDependencies.ts

```diff
@@ -72,13 +80,21 @@ export function pruneOrGetComponentDependencies(context: WLECleanerContext, prop
         // TODO track dependency
 
         // TODO prune default
+    } else if (propConfig.type === Type.ParticleEffect) {
+        // TODO
+    } else if (propConfig.type === Type.Array) {
+        // TODO
+    } else if (propConfig.type === Type.Record) {
+        // TODO
     } else if (propConfig.type !== Type.Object) {
-        let typeIDName: string | unknown = propConfig.type;
-        if (typeof typeIDName !== 'string') {
-            typeIDName = `<internal wle-cleaner ID (${String(typeIDName)})>`;
-        }
+        if (throwOnUnexpected) {
```

Comment:
i honestly don't know why this was added. this error exists to prevent projects from being corrupted. afaik the only way this can happen is if the editor bundle is broken, or if you're on a new WLE version that isn't supported by the cleaner, and you *really* don't want to be running the cleaner in these situations